### ISSUE TITLE
Expose celestial coordinate frame as attribute on NDCube

### ISSUE THREAD

**Issue by wtbarnes** [sunpy/ndcube#421]
It is often useful to know the celestial coordinate frame corresponding to a particular WCS, e.g. when overplotting a coordinate on a single slice of 3D data cube, where that slice corresponds to a particular dimension in time. It would be useful to expose this coordinate frame as an attribute on an `NDCube`. This can presumably be derived from the WCS using [`wcs_to_celestial_frame`](https://docs.astropy.org/en/stable/api/astropy.wcs.utils.wcs_to_celestial_frame.html#astropy.wcs.utils.wcs_to_celestial_frame).

**Comment by Cadair** [comment]
I agree this would be really useful but some things to consider are:

* `wcs_to_celestial_frame` is `astropy.wcs.WCS` specific and wouldn't work for gWCS or other APE14 compatible WCSes (an equivalent could easily be written for gwcs)
* What about other coordinate systems, should we provide a reference frame for spectral or temporal.
* What would this API look like, and how would it handle a cube without a celestial component?

**Comment by wtbarnes** [comment]
* You could have a `coordinate_frame` or `frame` attribute that looks at the type of WCS you have and then applies the appropriate translation from WCS to frame. Initially we `astropy.wcs.WCS` and gWCS could be supported and maybe provide some sort of entry point for translations from other WCSes
* I have no concept for what a reference frame for either of those two coordinate systems would look like.
* I mentioned this in the chat as well, but I wonder if it would be worth providing a special subclass that is specifically for cubes with two spatial (celestial) axes and a third non-celestial axis, be it time, wavelength, stokes, etc. The `frame` attribute would then only live on that subclass. This of course would be less appropriate for ndcube and more appropriate for say sunpy. This also definitely feeds into our discussion about the `Map` refactor.
